Docs: export customizations (#623)
telerik/blazor-docs: components/grid/export/events.md

---
title: Export Events
page_title: Grid - Export Events
description: Export to Excel the Grid for Blazor.
slug: grid-export-events
tags: telerik,blazor,grid,export,events
published: True
position: 15
---

# Export Events

You can customize the files exported to Excel and CSV by using the [OnBeforeExport](#onbeforeexport) and the [OnAfterExport](#onafterexport) events exposed to the `GridExcelExport` and `GridCsvExport` tags. 

#### In this article

* [OnBeforeExport](#onbeforeexport)
    * [For Excel Export](#for-excel-export)
    * [For CSV export](#for-csv-export)
    * [Examples](#examples)

* [OnAfterExport](#onafterexport)
    * [For Excel Export](#for-excel-export)
    * [For CSV export](#for-csv-export)
    * [Examples](#examples)

## OnBeforeExport

> The `OnBeforeExport` event fires after the user clicked the `ExcelExport` or `CsvExport` buttons, or by [programmatically invoking the export]({%slug grid-export-excel%}#programmatic-export-from-code). The event handler receives a `GridBeforeExcelExportEventArgs` and `GridBeforeCsvExportEventArgs` object, depending on the type of export, which provides the following fields:

Reviewer: ```suggestion
The `OnBeforeExport` event fires when the user clicks on a export button, or when [export is invoked programmatically]({%slug grid-export-excel%}#programmatic-export). The event handler receives a `GridBeforeExcelExportEventArgs` and `GridBeforeCsvExportEventArgs` object, depending on the type of export, which provides the following fields:
```

## For Excel Export

> * `Columns` - `List<ExcelExportableColumns` - a collection of all exportable columns in the Grid (the columns that have a defined `Field` and are visible). You can customize the following attributes of the Grid column before exporting it into Excel:

Reviewer: ```suggestion
* `Columns` - `List<ExcelExportableColumns` - a collection of all exportable columns in the Grid (the columns that have a defined `Field` and are visible). You can customize the following attributes of the Grid columns before exporting to Excel:
```

>     * `Width` - define the width of the column when exported to excel file.

Reviewer: ```suggestion
    * `Width` - define the width of the column **in pixels**.
```

    * `Title` - define the title of the column that would be shown in the header in excel file. 
    * `NumberFormat` - you can use the `NumberFormat` to provide an Excel-compatible number/date format

>     * `Field` - provides the data bound field of the column.

Reviewer: ```suggestion
    * `Field` - set the data bound field of the column.
```

    
    
* `Data` - `IEnumerable<object>` - you can assign a custom collection of data to be exported to excel such as the selected items in the Grid. 

> * `isCancelled` -  `bool` - you can cancel the OnBeforeExcel event by setting the `isCancelled` field to `true`.

Reviewer: field values normally don't change

```suggestion
* `isCancelled` -  `bool` - cancel the OnBeforeExcel event by setting the `isCancelled` property to `true`.
```

### For CSV Export

> * `Data` - `IEnumerable<object>` - you can assign a custom collection of data to be exported to excel/csv such as the selected items in the Grid. 

Reviewer: ```suggestion
* `Data` - `IEnumerable<object>` - assign a custom collection of data to be exported to CSV, for example only the selected items in the Grid. 
```

> * `isCancelled` -  `bool` - you can cancel the OnBeforeExcel event by setting the `isCancelled` field to `true`.

Reviewer: ```suggestion
* `isCancelled` -  `bool` - cancel the OnBeforeExcel event by setting the `isCancelled` property to `true`.
```

>note Unlike Excel, the CSV files do not have defined columns, so the `args.Columns` field will not be applicable when exporting the Grid to CSV.

### Examples

<div class="skip-repl"></div>
````Excel

> @* This example showcases the capabilities of the OnBeforeExport event when exporting the Grid to excel. *@

Reviewer: ```suggestion
@* This example shows the capabilities of the OnBeforeExport event when exporting the Grid to excel. *@
```

    </GridColumns>
</TelerikGrid>

@code {
    private async Task OnBeforCsvExport(GridBeforeCsvExportEventArgs args)
    {
        args.Data = SelectedItems;

        args.IsCancelled = false;
    }

    private IEnumerable<object> SelectedItems = Enumerable.Empty<object>();

    List<SampleData> GridData { get; set; }
    bool ExportAllPages { get; set; }

    protected override void OnInitialized()
    {
        GridData = Enumerable.Range(1, 100).Select(x => new SampleData
        {
            ProductId = x,
            ProductName = $"Product {x}",
            UnitsInStock = x * 2,
            Price = 3.14159m * x,
            Discontinued = x % 4 == 0,
            FirstReleaseDate = DateTime.Now.AddDays(-x)
        }).ToList();
    }

    public class SampleData
    {
        public int ProductId { get; set; }
        public string ProductName { get; set; }
        public int UnitsInStock { get; set; }
        public decimal Price { get; set; }
        public bool Discontinued { get; set; }
        public DateTime FirstReleaseDate { get; set; }
    }
}
````

## OnAfterExport

> The `OnAfterExport` event fires after the [OnBeforeExport](#onbeforeexport) event, and before the actual file is provided to the user. The event handler receives a `GridAfterExcelExportEventArgs` and `GridAfterCsvExportEventArgs` object, depending on the type of export, which provides the following fields:

Reviewer: ```suggestion
The `OnAfterExport` event fires after the [OnBeforeExport](#onbeforeexport) event, and before the actual file is provided to the user. The event handler receives a `GridAfterExcelExportEventArgs` or `GridAfterCsvExportEventArgs` object, depending on the type of export, which provides the following fields:
```
